pkg/asset/installconfig/azure/session.go

```diff
@@ -224,11 +226,14 @@ func newSessionFromCredentials(cloudEnv azureenv.Environment, credentials *Crede
 		AADEndpoint:  cloudEnv.ActiveDirectoryEndpoint,
 	}
 	c.Resource = cloudEnv.TokenAudience
-	authorizer, err := c.Authorizer()
+
+	cred, err := azidentity.NewClientSecretCredential(credentials.TenantID, credentials.ClientID, credentials.ClientSecret, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get client credentials authorizer")
+		return nil, errors.Wrap(err, "failed to get client credentials secret")
 	}
 
+	authorizer := azidext.NewTokenCredentialAdapter(cred, []string{azidext.DefaultManagementScope})
```

Comment:
I ran the azure stack CI test, but I'm pretty sure this is not going to work because the `DefaultManagementScope` is pointing to public Azure. Instead, we need to populate [DefaultAzureCredentialOptions](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#DefaultAzureCredentialOptions) with the appropriate values for the environment.